multiplier_analysis/nbconverted/1_get_multiplier_LV_coverage.py

```diff
@@ -49,265 +51,249 @@
-    
-gene_cov_matched = pd.DataFrame({'Proportion of significantly associated LVs covered': generic_cov_matched + specific_cov_matched,
-                         'gene type': ['generic']*len(ls_data_files) + ['specific']*len(ls_data_files) 
-                      })
+all_coverage = []
+for gene_label in dict_genes.keys():
+    merged_df = pd.DataFrame(
+        dict_nonzero_coverage[gene_label],
+        columns= ["nonzero LV coverage"]
+    ).merge(
+        pd.DataFrame(
+            dict_highweight_coverage[gene_label],
+            columns= ["highweight LV coverage"]
+        ),
+        left_index=True,
+        right_index=True
+    ) 
+    merged_df['gene type'] = gene_label
+    all_coverage.append(merged_df)
+
+all_coverage_df = pd.concat(all_coverage)
 
 
 # In[18]:
 
 
+all_coverage_df.head()
+
+
+# In[19]:
+
+
 # Plot coverage distribution given list of generic coverage, specific coverage
-print(generic_cov_matched)
-print(specific_cov_matched)
+nonzero_fig = sns.boxplot(
+    data=all_coverage_df, 
+    x='gene type',
+    y='nonzero LV coverage',
+    notch=True,
+    palette=['powderblue', 'grey']
+                         )
+plt.ylim(0, 700)
+nonzero_fig.set_xlabel("Gene Type",fontsize=14)
+nonzero_fig.set_ylabel(textwrap.fill("Number of LVs", width=30),fontsize=14)
+nonzero_fig.tick_params(labelsize=14)
+nonzero_fig.set_title("Number of LVs genes are present in", fontsize=16)
 
-import textwrap
-fig2 = sns.swarmplot(data=gene_cov_matched, 
-                  x='gene type', 
-                  y='Proportion of significantly associated LVs covered', 
-                  palette=['grey','powderblue'])
-fig2.set_xlabel("Gene Type",fontsize=14)
-fig2.set_ylabel(textwrap.fill("Proportion of pathway-associated LVs covered", width=30),fontsize=14)
-fig2.tick_params(labelsize=14)
-fig2.set_title("Coverage of pathway-associated LVs", fontsize=16)
 
+# In[20]:
+
+
+# Plot coverage distribution given list of generic coverage, specific coverage
+highweight_fig = sns.boxplot(data=all_coverage_df, 
+                             x='gene type',
+                             y='highweight LV coverage',
+                             notch=True,
+                             palette=['powderblue', 'grey']
+                            )
+plt.ylim(0, 700)
+highweight_fig.set_xlabel("Gene Type",fontsize=14)
+highweight_fig.set_ylabel(textwrap.fill("Number of LVs", width=30),fontsize=14)
+highweight_fig.tick_params(labelsize=14)
+highweight_fig.set_title("Number of LVs genes contribute highly to", fontsize=16)
+
+
+# ## Get LVs that generic genes are highly weighted in
+# 
+# Currently, I get high weight genes if the gene weight > quantile threshold for that specific LV. Each LV has a different threshold based on its own distribution. So each LV has the same number of high weight genes. 
+# 
+# Should I have normalized the data per LV before calculating which genes are high weight per LV?
```

Comment:
Re. this:

> Should I have normalized the data per LV before calculating which genes are high weight per LV?

I think it would be worth trying. I guess what you would do is first normalize the columns of the MultiPLIER matrix (using, say, [StandardScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html)), then pick a cutoff and use it for all the columns.

My guess is that this will give you very similar results to what you have, though. If what you get is substantially different, that would mean some of the LV weight distributions are quite skewed (since using quantiles will "correct" for this to some degree, and normalization won't really). I don't necessarily think there's anything wrong with the way you're doing things now (or I think you can make a strong argument for it being correct, anyway), but trying the standardization approach as well might help you to put your results into context more.

Reply:
One thing Casey and I discussed was that it would be interesting to look at what LVs do these generic genes contribute a lot to. 

This question arose because I didn't think I could select LVs using the quantile approach since each LV had the same number of genes. At the time, I thought they all had the same number of high weight **generic genes**, but i realized today that they each have the same number of high weight genes **not necessarily generic** So I think I can just look at the proportion of high weight genes that are generic vs other to try to pull out LVs to look at - as in try to pull out LVs that have a higher proportion of generic high weight genes and look at the pathways associated with those LVs